app/src/main/java/com/jerboa/Utils.kt

```diff
@@ -238,64 +239,135 @@ data class InstantScores(
     val downvotes: Int,
 )
 
-data class CommentNodeData(
-    val commentView: CommentView,
-    // Must use a SnapshotStateList and not a MutableList here, otherwise changes in the tree children won't trigger a UI update
-    val children: SnapshotStateList<CommentNodeData>?,
-    var depth: Int,
+data class MissingCommentView(
+    val commentId: Int,
+    val path: String,
 )
 
+sealed class CommentNodeData(
+    val depth: Int,
+    // Must use a SnapshotStateList and not a MutableList here, otherwise changes in the tree children won't trigger a UI update
+    val children: SnapshotStateList<CommentNodeData> = mutableStateListOf(),
+    var parent: CommentNodeData? = null,
+) {
+    abstract fun getId(): Int
+    abstract fun getPath(): String
+}
+
+class CommentNode(
+    val commentView: CommentView,
+    depth: Int,
+    children: SnapshotStateList<CommentNodeData> = mutableStateListOf(),
+    parent: CommentNodeData? = null,
+) : CommentNodeData(depth, children, parent) {
+    override fun getId() = commentView.comment.id
+    override fun getPath() = commentView.comment.path
+}
+
+class MissingCommentNode(
+    val missingCommentView: MissingCommentView,
+    depth: Int,
+    children: SnapshotStateList<CommentNodeData> = mutableStateListOf(),
+    parent: CommentNodeData? = null,
+) : CommentNodeData(depth, children, parent) {
+    override fun getId() = missingCommentView.commentId
+    override fun getPath() = missingCommentView.path
+}
+
 fun commentsToFlatNodes(
     comments: List<CommentView>,
-): ImmutableList<CommentNodeData> {
-    return comments.map { c -> CommentNodeData(commentView = c, children = null, depth = 0) }.toImmutableList()
+): ImmutableList<CommentNode> {
+    return comments.map { c -> CommentNode(c, depth = 0) }.toImmutableList()
 }
 
+/**
+ * This function takes a list of comments and builds a tree from it
+ *
+ * In commentView it should be giving a id of the root comment
+ * Else it would generate a
+ */
 fun buildCommentsTree(
     comments: List<CommentView>,
-    isCommentView: Boolean,
+    rootCommentId: Int?, // If it's in CommentView, then we need to know the root comment id
 ): ImmutableList<CommentNodeData> {
+    val isCommentView = rootCommentId != null
+
     val map = LinkedHashMap<Number, CommentNodeData>()
     val firstComment = comments.firstOrNull()?.comment
 
-    val depthOffset = if (!isCommentView) {
-        0
+    val depthOffset = if (isCommentView && firstComment != null) {
+        getCommentIdDepthFromPath(firstComment.path, rootCommentId!!)
     } else {
-        getDepthFromComment(firstComment) ?: 0
```

Comment:
This logic was wrong and caused the depth of the commentTree to be reversed.

Because this incorrectly assumes that the first comment has the lowest depth. (This is not always case such as when you use the parent_id due to a bug in 0.18.3) Thus causing the list to be generated with negative depth, This doesn't a problem (by sheer luck?) because the color handles this negative depth because it uses modulo and the calculateCommentOffset handles this correctly because it uses abs.

It does it correctly now because it gets the depth of the root comment. (which is always the lowest)

see https://lemmy.one/api/v3/comment/list?max_depth=6&parent_id=3355587&sort=Hot&type_=All&auth=REDACTED

Before
![image](https://github.com/dessalines/jerboa/assets/67873169/38d007a9-71d7-4797-aff8-b697010fa1bd)

After:
![image](https://github.com/dessalines/jerboa/assets/67873169/dd663e1b-b35e-4c57-84f2-315b4e6657de)